cmd/alertmanager/main.go

```diff
@@ -180,10 +182,14 @@ func run() int {
 	}
 
 	var (
-		configFile      = kingpin.Flag("config.file", "Alertmanager configuration file name.").Default("alertmanager.yml").String()
-		dataDir         = kingpin.Flag("storage.path", "Base path for data storage.").Default("data/").String()
-		retention       = kingpin.Flag("data.retention", "How long to keep data for.").Default("120h").Duration()
-		alertGCInterval = kingpin.Flag("alerts.gc-interval", "Interval between alert GC.").Default("30m").Duration()
+		configFile = kingpin.Flag("config.file", "Alertmanager configuration file name.").Default("alertmanager.yml").String()
+		dataDir    = kingpin.Flag("storage.path", "Base path for data storage.").Default("data/").String()
+		retention  = kingpin.Flag("data.retention", "How long to keep data for.").Default("120h").Duration()
+
+		alertGCInterval      = kingpin.Flag("alerts.gc-interval", "Interval between alert GC.").Default("30m").Duration()
+		alertStorageProvider = kingpin.Flag("alerts.storage.provider", "Alert storage \"mem\" or \"etcd\".").Default("mem").String()
```

Comment:
Q: Is `etcd` storage provider indicative of adding `etcd` alongside `mem` behavior -- but importantly not indicative of subtracting `mem` functionality entirely.

Reply:
etcd storage provider is a clone of mem storage provider, with a few lines of code added to it.  There's not much diff between mem.go and etcd.go.  if the code organization were refactored, this code duplicity could go away.